crates/forge/src/test_filter.rs

```diff
@@ -43,7 +48,9 @@ impl TestsFilter {
             IgnoredFilter::NotIgnored
         };
 
-        let name_filter = if exact_match {
+        let name_filter = if rerun_failed {
+            NameFilter::RerunFailed
+        } else if exact_match {
```

Comment:
It shouldn't override the filtering with names, those should be independent. I. e., I would expect `snforge test --rerun-failed xyz` to only run tests that failed the last time with `xyz` in their logical path

Reply:
IMO it's a separate feature, it demands merging filters

Reply:
Yea, those are separate features, that's why they shouldn't be merged. Filtering tests based on `--rerun-failed`should be independent from filtering based on `[TEST_FILTER]` and `--exact`. It would be really confusing if  we would change the behaviour of filtering by name when `--rerun-failed` flag is specified